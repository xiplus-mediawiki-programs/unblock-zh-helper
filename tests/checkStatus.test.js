import { flushPromises, mount } from '@vue/test-utils';
import App from '../src/App.vue';
import createMyI18n from '../src/lang';
import { expect, test, describe } from 'vitest';

describe('form', async () => {
  const wrapper = mount(App, {
    global: {
      plugins: [createMyI18n('zh-hant')],
    },
  });
  await wrapper.setData({
    mailOptionsVariant: 'zh-hant',
  });

  test('status 1 normalized', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      username: 'xiplus',
      normalizedUsername: 'Xiplus',
    });
    expect(wrapper.find('#uzh-req-status').findAll('li')[0].wrapperElement).toMatchSnapshot();
  });

  test('status 2 account can be create', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      normalizedUsername: 'Xiplus',
      usernameStatus: wrapper.vm.ACCST_NOT_EXISTS,
    });
    expect(wrapper.find('#uzh-req-status').findAll('li')[1].wrapperElement).toMatchSnapshot();
  });

  test('status 2 account not exists', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      inputCreateAccount: false,
      normalizedUsername: 'Xiplus',
      usernameStatus: wrapper.vm.ACCST_NOT_EXISTS,
    });
    expect(wrapper.find('#uzh-req-status').findAll('li')[1].wrapperElement).toMatchSnapshot();
  });

  test('status 3 account needs local', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      normalizedUsername: 'Xiplus',
      usernameStatus: wrapper.vm.ACCST_NEEDS_LOCAL,
    });
    expect(wrapper.find('#uzh-req-status').findAll('li')[1].wrapperElement).toMatchSnapshot();
  });

  test('status 4 account exists req acct', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      normalizedUsername: 'Xiplus',
      usernameStatus: wrapper.vm.ACCST_EXISTS,
    });
    expect(wrapper.find('#uzh-req-status').findAll('li')[1].wrapperElement).toMatchSnapshot();
  });

  test('status 5 account exists not req acct', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      inputCreateAccount: false,
      normalizedUsername: 'Xiplus',
      usernameStatus: wrapper.vm.ACCST_EXISTS,
    });
    expect(wrapper.find('#uzh-req-status').findAll('li')[1].wrapperElement).toMatchSnapshot();
  });

  test('status 6 account banned', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      inputCreateAccount: false,
      normalizedUsername: 'Xiplus',
      usernameStatus: wrapper.vm.ACCST_BANNED,
    });
    expect(wrapper.find('#uzh-req-status').findAll('li')[1].wrapperElement).toMatchSnapshot();
  });

  test('status 7 account blocked', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      normalizedUsername: 'Xiplus',
      accountBlocked: true,
      accountBlockBy: 'BlockAdmin',
      accountBlockReason: 'Vandalism',
    });
    expect(wrapper.find('#uzh-req-status').findAll('li')[1].wrapperElement).toMatchSnapshot();
  });

  test('status 8 ip blocked proxy', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      ipChecked: true,
      ip: '1.2.3.4',
      blocked: true,
      blockReason: '{{Blocked proxy}}',
    });
    expect(wrapper.find('#uzh-req-status').findAll('li')[0].wrapperElement).toMatchSnapshot();
  });

  test('status 8 ip blocked other', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      ipChecked: true,
      ip: '1.2.3.4',
      blocked: true,
      blockReason: 'Vandalism',
    });
    expect(wrapper.find('#uzh-req-status').findAll('li')[0].wrapperElement).toMatchSnapshot();
  });

  test('status 9 ip not blocked', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      ipChecked: true,
      ip: '1.2.3.4',
      blocked: false,
    });
    expect(wrapper.find('#uzh-req-status').findAll('li')[0].wrapperElement).toMatchSnapshot();
  });

  test('status 10 acct has ipbe', async () => {
    await wrapper.vm.resetForm();
    await wrapper.setData({
      accountHasIpbe: true,
      normalizedUsername: 'Xiplus',
    });
    expect(wrapper.find('#uzh-req-status').findAll('li')[1].wrapperElement).toMatchSnapshot();
  });
});
