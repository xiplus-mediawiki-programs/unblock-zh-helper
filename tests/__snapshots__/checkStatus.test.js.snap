// Vitest Snapshot v1

exports[`check status > status 1 normalized 1`] = `
<li
  class="uzh-status-info"
>
  使用者名稱被正規化為「Xiplus」
</li>
`;

exports[`check status > status 2 account can be create 1`] = `
<li
  class="uzh-status-success"
>
  <span>
    帳號可以建立（
    <a
      href="https://www.google.com/search?q=Xiplus"
      target="_blank"
    >
      Google搜尋
    </a>
    ）
  </span>
</li>
`;

exports[`check status > status 2 account not exists 1`] = `
<li
  class="uzh-status-error"
>
  帳號不存在
</li>
`;

exports[`check status > status 3 account needs local 1`] = `
<li
  class="uzh-status-error"
>
  <span>
    需要強制建立本地帳號（
    <a
      href="/w/index.php?title=Special:CentralAuth&target=Xiplus"
      target="_blank"
    >
      全域帳號
    </a>
    ）
  </span>
</li>
`;

exports[`check status > status 4 account exists req acct 1`] = `
<li
  class="uzh-status-error"
>
  <span>
    帳號已被註冊（
    <a
      href="/w/index.php?title=Special:CentralAuth&target=Xiplus"
      target="_blank"
    >
      全域帳號
    </a>
    、
    <a
      href="/w/index.php?title=Special:Log&type=rights&page=User%3AXiplus"
      target="_blank"
    >
      權限日誌
    </a>
    ）
  </span>
</li>
`;

exports[`check status > status 5 account exists not req acct 1`] = `
<li
  class="uzh-status-success"
>
  <span>
    帳號已被註冊（
    <a
      href="/w/index.php?title=Special:CentralAuth&target=Xiplus"
      target="_blank"
    >
      全域帳號
    </a>
    、
    <a
      href="/w/index.php?title=Special:Log&type=rights&page=User%3AXiplus"
      target="_blank"
    >
      權限日誌
    </a>
    ）
  </span>
</li>
`;

exports[`check status > status 6 account banned 1`] = `
<li
  class="uzh-status-error"
>
  帳號不存在
</li>
`;

exports[`check status > status 7 account blocked 1`] = `
<li
  class="uzh-status-error"
>
  <span>
    <b>
      警告：帳號被BlockAdmin封鎖，原因為：Vandalism
    </b>
    （
    <a
      href="/w/index.php?title=Special:Log/block&page=User%3AXiplus"
      target="_blank"
    >
      封鎖日誌
    </a>
    ）
  </span>
</li>
`;

exports[`check status > status 8 ip blocked other 1`] = `
<li
  class="uzh-status-error"
>
  <span>
    <span>
      IP已被封鎖，原因為：Vandalism
    </span>
    （
    <a
      href="/w/index.php?title=Special:BlockList&wpTarget=1.2.3.4"
      target="_blank"
    >
      檢查
    </a>
    ）
  </span>
</li>
`;

exports[`check status > status 8 ip blocked proxy 1`] = `
<li
  class="uzh-status-success"
>
  <span>
    <span>
      IP已被封鎖，原因為：{{Blocked proxy}}
    </span>
    （
    <a
      href="/w/index.php?title=Special:BlockList&wpTarget=1.2.3.4"
      target="_blank"
    >
      檢查
    </a>
    ）
  </span>
</li>
`;

exports[`check status > status 9 ip not blocked 1`] = `
<li
  class="uzh-status-error"
>
  <span>
    申請人給定的IP未被封鎖（
    <a
      href="/w/index.php?title=Special:BlockList&wpTarget=1.2.3.4"
      target="_blank"
    >
      檢查
    </a>
    ）
  </span>
</li>
`;

exports[`check status > status 10 acct has ipbe 1`] = `
<li
  class="uzh-status-error"
>
  <span>
    使用者已擁有IP封鎖豁免權（
    <a
      href="/w/index.php?title=Special:Log&type=rights&page=User%3AXiplus"
      target="_blank"
    >
      權限日誌
    </a>
    ）
  </span>
</li>
`;
